packages/syft/src/syft/core/node/common/managers/bin_obj_manager.py

```diff
@@ -0,0 +1,168 @@
+# stdlib
+from typing import Iterable
+from typing import KeysView
+from typing import List
+from typing import Optional
+
+# third party
+from nacl.encoding import HexEncoder
+from nacl.signing import VerifyKey
+from sqlalchemy.orm.decl_api import DeclarativeMeta
+from sqlalchemy.orm.session import Session
+from torch import Tensor
+
+# syft absolute
+import syft
+from syft.core.common.group import VERIFYALL
+from syft.core.common.uid import UID
+from syft.core.store import ObjectStore
+from syft.core.store.storeable_object import StorableObject
+
+ENCODING = "UTF-8"
+
+
+def create_storable(
+    _id: UID, data: Tensor, description: str, tags: Iterable[str]
+) -> StorableObject:
+    obj = StorableObject(id=_id, data=data, description=description, tags=tags)
+
+    return obj
+
+
+def storable_to_dict(storable_obj: StorableObject) -> dict:
+    _dict = {}
+    _dict["tags"] = storable_obj.tags
+    _dict["description"] = storable_obj.description
+    # Serialize nacl Verify Keys Structure
+    _dict["read_permissions"] = {
+        key.encode(encoder=HexEncoder).decode("utf-8"): None
+        for key in storable_obj.read_permissions.keys()
+    }
+    return _dict
+
+
+class BinObjectManager(ObjectStore):
+    def __init__(
+        self,
+        db_session: Session,
+        bin_obj_table: DeclarativeMeta,
+        bin_obj_metadata_table: DeclarativeMeta,
+    ) -> None:
+        self.db_session = db_session
+        self.bin_obj_table = bin_obj_table
+        self.bin_obj_metadata_table = bin_obj_metadata_table
+
+    def get_object(self, key: UID) -> Optional[StorableObject]:
+        try:
+            return self.__getitem__(key)
+        except KeyError as e:  # noqa: F841
```

Comment:
can we actually get the logging of the error so we don't hide unwanted behaviour?

Reply:
Will rewrite most of this class next.